src/state/swap/reducer.ts

```diff
@@ -74,13 +87,26 @@ export default createReducer<SwapState>(initialState, builder =>
       }
     })
     .addCase(typeInput, (state, { payload: { field, typedValue } }) => {
+      let loading = false
+      if (
+        state[Field.INPUT].currencyId !== '' &&
+        state[Field.OUTPUT].currencyId !== '' &&
+        state.typedValue !== typedValue &&
+        Number(typedValue ?? 0) > 0
+      ) {
+        loading = true
+      }
       return {
         ...state,
         independentField: field,
         typedValue,
+        loading,
       }
     })
     .addCase(setRecipient, (state, { payload: { recipient } }) => {
       state.recipient = recipient
     })
+    .addCase(setLoading, (state, { payload: loading }) => {
+      state.loading = loading
+    })
```

Comment:
Now the implementation is a middle step since the eco-router was sync earlier. Once this is merged I can revisit and clean up the loading a bit. You dont need to update the loading since I am checking the output of In and Out  dexes length. 

Will use this loading once it is merged and it will be correct way to get the loading.